Source/Icebreaker/Bot/IcebreakerBot.cs

```diff
@@ -226,22 +233,26 @@ private async Task HandleMessageActivityAsync(ITurnContext turnContext, Cancella
             try
             {
                 var activity = turnContext.Activity;
-                var senderAadId = activity.From.AadObjectId;
+                var senderId = activity.From.Id;
                 var tenantId = activity.GetChannelData<TeamsChannelData>().Tenant.Id;
 
-                if (string.Equals(activity.Text, MatchingActions.OptOut, StringComparison.InvariantCultureIgnoreCase))
+                if (!string.IsNullOrEmpty(activity.ReplyToId) && (activity.Value != null) && ((JObject)activity.Value).HasValues)
+                {
+                    await this.OnProfileSaveAsync(activity, turnContext, cancellationToken).ConfigureAwait(false);
+                }
+                else if (string.Equals(activity.Text, MatchingActions.OptOut, StringComparison.InvariantCultureIgnoreCase))
                 {
                     // User opted out
-                    this.telemetryClient.TrackTrace($"User {senderAadId} opted out");
+                    this.telemetryClient.TrackTrace($"User {senderId} opted out");
 
                     var properties = new Dictionary<string, string>
                     {
-                        { "UserAadId", senderAadId },
+                        { "UserAadId", senderId },
```

Comment:
If we are changing this to the bot framework id this shouldn't be marked as "aad" id. These are two separate systems with two different identifiers for the user.